File: apps/android/feature/canvas/domain/src/main/java/com/scribblefit/feature/canvas/domain/usecase/ProcessScribbleUseCase.kt
package com.scribblefit.feature.canvas.domain.usecase

import com.scribblefit.feature.scribble.domain.ScribbleRepository
import kotlinx.coroutines.CoroutineDispatcher
import kotlinx.coroutines.withContext

class ProcessScribbleUseCase(
    private val scribbleRepository: ScribbleRepository,
    private val coroutineDispatcher: CoroutineDispatcher
) {
    suspend fun execute(rawText: String) = withContext(coroutineDispatcher) {
        scribbleRepository.enqueueScribble(rawText)
    }
}

File: apps/ios/ScribbleFit/Sources/Features/Canvas/Domain/UseCases/ProcessScribbleUseCase.swift
import Foundation

public final class ProcessScribbleUseCase: Sendable {
    private let scribbleRepository: any ScribbleRepository

    public init(scribbleRepository: any ScribbleRepository) {
        self.scribbleRepository = scribbleRepository
    }

    public func execute(rawText: String) async {
        await scribbleRepository.enqueueScribble(rawText: rawText)
    }
}
